text/trace/0170-sampling-probability.md

```diff
@@ -0,0 +1,677 @@
+collection at scale.  Dapper's early Sampler algorithm, referred to as
+an "inflationary" approach (although not published in the paper), is
+reproduced here.
+
+This kind of Sampler allows non-root spans in a trace to raise the
+probability of tracing, using a conditional probability formula shown
+below.  Traces produced in this way are complete sub-trees, not
+necessarily complete.  This technique is successful especially in
+systems where a high-throughput service on occasion calls a
+low-throughput service.  Low-throughput services are meant to inflate
+their sampling probability.
+
+The use of this technique requires propagating the head inclusion
+probability (as discussed for the `Parent` sampler) of the incoming
+Context and whether it was sampled, in order to calculate the
+probability of starting to sample a new "sub-root" in the trace.
+
+Using standard notation for conditional probability, `P(x)` indicates
+the probability of `x` being true, and `P(x|y)` indicates the
+probability of `x` being true given that `y` is true.  The axioms of
+probability establish that:
+
+```
+P(x)=P(x|y)*P(y)+P(x|not y)*P(not y)
+```
+
+The variables are:
+
+- **`H`**: The head inclusion probability of the parent context that
+  is in effect, independent of whether the parent context was sampled
+- **`I`**: The inflationary sampling probability for the span being
+  started.
+- **`D`**: The decision probability for whether to start a new sub-root.
+
+This Sampler cannot lower sampling probability, so if the new span is
+started with `H >= I` or when the context is already sampled, no new
+sampling decisions are made.  If the incoming context is already
+sampled, the adjusted count of the new span is `1/H`.
+
+Assuming `H < I` and the incoming context was not sampled, we have the
+following probability equations:
+
+```
+P(span sampled) = I
+P(parent sampled) = H
+P(span sampled | parent sampled) = 1
+P(span sampled | parent not sampled) = D
+```
+
+Using the formula above, 
+
+```
+I = 1*H + D*(1-H)
+```
+
+solve for D:
+
+```
+D = (I - H) / (1 - H)
+```
+
+Now the Sampler makes a decision with probability `D`.  Whether the
+decision is true or false, propagate `I` as the new head inclusion
+probability.  If the decision is true, begin recording a sub-rooted
+trace with adjusted count `1/I`.
+
+## Proposed specification text
+
+The following text will be added to the semantic conventions for
+recording the Sampler name and adjusted count (if known) as
+OpenTelemetry Span attributes.
+
+```
+# Semantic conventions for Sampled spans
+
+This document defines how to describe an what sampling was performed
+when recording a span that has had sampling logic applied.
+
+Span sampling attributes support computing metrics about spans that
+are part of a sampled trace from knowing their sampling inclusion
+probability.
+
+The _adjusted count_ of a span is defined as follows:
+
+- Adjusted count equals zero when inclusion probability equals zero
```

Comment:
The 0 value for adjusted count is meaningful and allows recording non-probabilistically selected Spans while performing probability sampling. In an earlier draft of #148 I had proposed a convention for supporting multiple independent probability samples, in which case -- when more than one Sampler may cause a Span to be recorded -- a zero value for adjusted count should be used to signify a value that is recorded but not selected in a particular sample.

I removed the detail about supporting multiple samples because OTel doesn't need to specify how to do more than one mechanism. The mechanism I believe OTel should require is the system-wide sample, which is the sample of all traces you get by default when turning on an OTel SDK. Thus, I am proposing a single set of `sampler.name` and `sampler.adjusted_count` attributes. Vendors and systems that want to use more than one sampler may end up recording spans with `sampler.adjusted_count=0` if they are selected by alternate samplers but not included in the system-wide sample.

Reply:
I have added a sentence to explain why zero adjusted count is useful and a sentence to explain the connection between expected-value and probability.